stm/stm.ml

```diff
@@ -269,7 +270,10 @@ module VCS : sig
 
   type vcs = (branch_type, transaction, vcs state_info, box) Vcs_.t
 
-  val init : id -> unit
+  val init : stm_doc_type -> id -> unit
+  (* val get_type : unit -> stm_doc_type *)
+  val is_interactive : unit -> [`Yes | `No | `Shallow]
```

Comment:
remove `Shallow` (polymorphic variants do have subtyping)

Reply:
Noted, thanks!

Reply:
I do get an error here; in fact I remember I tried to do that, not sure what is going on, I propose we leave this like this for now.